crates/storage/provider/src/providers/database/provider.rs

```diff
@@ -736,20 +735,66 @@ impl<'this, TX: DbTxMut<'this> + DbTx<'this>> DatabaseProvider<'this, TX> {
         Ok(())
     }
 
-    /// Load last shard and check if it is full and remove if it is not. If list is empty, last
+    /// Load shard and check if it is full and remove if it is not. If list is empty, last
```

Comment:
I'm not very familiar with shards logic, but do we still check if the shard is full in this function?

Reply:
Yep, makes sense. So something like this?
```suggestion
    /// Load shard and remove it. If list is empty, last
```